tests/nnstreamer_if/unittest_if.cc

```diff
@@ -691,6 +691,102 @@ TEST (tensor_if_appsrc, action_1)
   g_free (str_pipeline);
 }
 
+/**
+ * @brief custom callback function
+ */
+gboolean tensor_if_custom_cb (const GstTensorsInfo *info,
+    const GstTensorMemory * input, gboolean * result) {
+  gint *output, i;
+
+  if (!info || !input || !result)
+    return FALSE;
+
+  output = (gint *) input[1].data;
+  *result = TRUE;
+
+  for (i = 0; i < 48; i++) {
+    if (test_frames[1][i] != output[i]) {
+      *result = FALSE;
+      break;
+    }
+  }
+
+  return TRUE;
+}
+
+/**
+ * @brief Test behavior: custom callback
+ */
+TEST (tensor_if_custom, custom_0)
+{
+  GstBuffer *buf_0, *buf_1;
+  GstMemory *mem;
+  GstMapInfo info;
+  GstElement *appsrc_handle, *sink_handle, *tif_handle;
+  gint i, idx;
+
+  gchar *str_pipeline = g_strdup
+      ("appsrc name=appsrc ! other/tensors,num_tensors=2,dimensions=(string)3:4:2:2.3:4:2:2, types=(string)int32.int32,framerate=(fraction)0/1 ! "
+       "tensor_if name=tif compared_value=CUSTOM then=TENSORPICK then-option=0 else=TENSORPICK else-option=1 "
+            "tif.src_0 ! queue ! tensor_sink name=sink_true async=false "
+            "tif.src_1 ! queue ! tensor_sink name=sink_false async=false");
+
+  GstElement *pipeline = gst_parse_launch (str_pipeline, NULL);
+  EXPECT_NE (pipeline, nullptr);
+
+  appsrc_handle = gst_bin_get_by_name (GST_BIN (pipeline), "appsrc");
+  EXPECT_NE (appsrc_handle, nullptr);
+
+  tif_handle = gst_bin_get_by_name (GST_BIN (pipeline), "tif");
+  EXPECT_NE (tif_handle, nullptr);
+
+  sink_handle = gst_bin_get_by_name (GST_BIN (pipeline), "sink_true");
+  EXPECT_NE (sink_handle, nullptr);
+
+  g_signal_connect (sink_handle, "new-data", (GCallback) new_data_cb, (gpointer) &idx);
+  gst_object_unref (sink_handle);
+
+  sink_handle = gst_bin_get_by_name (GST_BIN (pipeline), "sink_false");
+  EXPECT_NE (sink_handle, nullptr);
+
+  g_signal_connect (sink_handle, "new-data", (GCallback) new_data_cb, (gpointer) &idx);
+  gst_object_unref (sink_handle);
+
+  nnstreamer_if_custom_register (tif_handle, tensor_if_custom_cb, NULL);
+
+  buf_0 = gst_buffer_new ();
```

Comment:
Is there no problem without unref like `gst_buffer_unref (buf_0);` at the end of this test case?

Reply:
Yes. `gst_app_src_push_buffer` takes ownership of the buffer. So don't need to unref buffer.